### PR TITLE
docs/OSUpgrades.md: Update/reword

### PR DESCRIPTION
**Pull request by cgwalters**
Various updates/clarifications here.  (If anyone wants to
do nontrivial edits, feel free to toss this into a hackmd.io
doc and we can collaborate!)

### CODE REVIEW

File: docs/OSUpgrades.md [hunk 1]
```diff
@@ -14,83 +14,98 @@ These bootimages are built using [coreos-assembler](https://github.com/coreos/co
 Today, [the installer](https://github.com/openshift/installer/) pins the "bootimages"
 it uses, and released installers also pin the release image.  As noted above,
 release images contain `machine-os-content`, which can be a *different*
-RHCOS version.
+RHCOS version.  You can find the installer-pinned bootimage in e.g. [this file](https://github.com/openshift/installer/blob/release-4.4/data/data/rhcos.json).
```
Comment (yuqi-zhang) [inline docs/OSUpgrades.md:17]: maybe link to master instead of the release-4.4 branch

File: docs/OSUpgrades.md [hunk 2]
```diff
@@ -14,83 +14,98 @@ These bootimages are built using [coreos-assembler](https://github.com/coreos/co
...
-The next time the admin does an `oc adm upgrade`, if a new `machine-os-content`
-is provided in the release image, it will be rolled out to the masters
-and workers.
+Today mostly because of [SELinux reasons](https://bugzilla.redhat.com/show_bug.cgi?id=1839065) the
+MCD copies itself to the host, and runs as part of the host context.
+Then it provides updated content to `rpm-ostreed.service`, which is already
+a daemon running on the host.
+
+The MCD tries to watch the systemd journal for relevant service and proxy logs,
+so you *should* be able to `oc -n openshift-machine-config-operator logs pod/machine-config-daemon-...`
```
Comment (yuqi-zhang) [inline docs/OSUpgrades.md:106]: should we specify `oc -n openshift-machine-config-operator logs -c machine-config-daemon pod/machine-config-daemon-...` for clarity

File: docs/OSUpgrades.md (hunk 1 above)
Comment (kikisdeliveryservice) [reply docs/OSUpgrades.md:17]: agree

File: docs/OSUpgrades.md (hunk 2 above)
Comment (kikisdeliveryservice) [reply docs/OSUpgrades.md:106]: agree

File: docs/OSUpgrades.md (hunk 1 above)
Comment (cgwalters) [reply docs/OSUpgrades.md:17]: But that file is highly likely to go away after https://github.com/openshift/enhancements/pull/201 - seems more useful to have an old but valid link.

File: docs/OSUpgrades.md [hunk 3]
```diff
@@ -14,83 +14,98 @@ These bootimages are built using [coreos-assembler](https://github.com/coreos/co
...
 The OSTree format is an image format designed for in-place operating system updates; it operates
 at the filesystem level (like container images) but (unlike container runtimes) has
 tooling to manage things such as the bootloader and handling persistence of `/etc` and `/var`.
 
-On top of the OSTree format, Red Hat Enterprise Linux CoreOS uses [pivot](https://github.com/openshift/pivot/)
-which is a "glue layer" that handles the encapsulation of the `machine-os-content` (oscontainer)
-with the OSTree repository.
+The reason we wrap an OSTree commit inside a container image is so that
+the release image encapsulates basically everything about a cluster (except the bootimage).
+This makes it easy to e.g. mirror updates offline.
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply docs/OSUpgrades.md:31]: It's not *just* about mirroring though.  Having a single artifact means that for example in CI we can test *exactly one thing* - today the RHCOS dependency tends to "leak" outside of the release image.

File: docs/OSUpgrades.md [hunk 4]
```diff
@@ -14,83 +14,98 @@ These bootimages are built using [coreos-assembler](https://github.com/coreos/co
...
+After a node (whether control plane or worker) has joined the cluster, the MCO
+takes over.  Previously, each individual node was running systemd units;
+now changes are coordinated via the operator.
+
+When the administrator starts an `oc adm upgrade`, if a new `machine-os-content`
+is provided in the release image, it will be rolled out to the control plane
+and workers.
 
 Every change now will be managed by a `machineconfigpool`, ensuring
 that only 1 machine at a time is changed (via `maxUnavailable: 1` default).
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply docs/OSUpgrades.md:97]: I think the target audience for this doc is someone who understands operating systems and Kubernetes at a high level - they're just looking for an explanation of the radically novel and unique thing we've done to combine them :smile: 

So we should expect them to know how to `oc get machineconfigpool`.

**Comment by cgwalters** [comment]
Pushed an update for various bits - in the future I'll probably put up a hackmd.io so we can do this more collaboratively.

**Comment by cgwalters** [comment]
Any blockers for landing this?

**Comment by yuqi-zhang** [comment]
/override ci/prow/e2e-aws-workers-rhel7

no reason to have this continue running for doc changes

**Pull request merged**